### PR TITLE
[Runtime] Add TVM_DLL to memory manager functions

### PR DESCRIPTION
**Pull request by MasterJH5574**
This PR adds TVM_DLL to memory manager runtime functions for windows builds.

### CODE REVIEW

**Review by tqchen — APPROVED** [review]

File: include/tvm/runtime/memory/memory_manager.h [hunk 1]
```diff
@@ -140,12 +140,12 @@ class StorageObj : public Object {
...
   /*! \brief Allocate an NDArray from a given piece of storage. */
-  NDArray AllocNDArray(int64_t offset, ShapeTuple shape, DLDataType dtype);
+  TVM_DLL NDArray AllocNDArray(int64_t offset, ShapeTuple shape, DLDataType dtype);
 
   /*! \brief The deleter for an NDArray when allocated from underlying storage. */
-  static void Deleter(Object* ptr);
+  TVM_DLL static void Deleter(Object* ptr);
 
-  ~StorageObj() {
+  TVM_DLL ~StorageObj() {
```
Comment (tqchen) [inline include/tvm/runtime/memory/memory_manager.h:148]: for definition in the header, we should not provide TVM_DLL

File: include/tvm/runtime/memory/memory_manager.h [hunk 2]
```diff
@@ -140,12 +140,12 @@ class StorageObj : public Object {
   Buffer buffer;
 
   /*! \brief Allocate an NDArray from a given piece of storage. */
-  NDArray AllocNDArray(int64_t offset, ShapeTuple shape, DLDataType dtype);
+  TVM_DLL NDArray AllocNDArray(int64_t offset, ShapeTuple shape, DLDataType dtype);
 
   /*! \brief The deleter for an NDArray when allocated from underlying storage. */
-  static void Deleter(Object* ptr);
+  TVM_DLL static void Deleter(Object* ptr);
```
Comment (tqchen) [inline include/tvm/runtime/memory/memory_manager.h:146]: This is not used by external users, so it is not necessary to expose it

File: include/tvm/runtime/memory/memory_manager.h [hunk 3]
```diff
@@ -57,17 +57,17 @@ struct Buffer {
 
 class Allocator {
  public:
-  explicit Allocator(AllocatorType type) : type_(type) {}
-  virtual ~Allocator() = default;
+  TVM_DLL explicit Allocator(AllocatorType type) : type_(type) {}
+  TVM_DLL virtual ~Allocator() = default;
```
Comment (tqchen) [inline include/tvm/runtime/memory/memory_manager.h:61]: this is defined in the header (=default), no need to add TVM_DLL

File: include/tvm/runtime/memory/memory_manager.h [hunk 4]
```diff
@@ -57,17 +57,17 @@ struct Buffer {
 
 class Allocator {
  public:
-  explicit Allocator(AllocatorType type) : type_(type) {}
-  virtual ~Allocator() = default;
+  TVM_DLL explicit Allocator(AllocatorType type) : type_(type) {}
```
Comment (tqchen) [inline include/tvm/runtime/memory/memory_manager.h:60]: This is not necessary as it is defined in the header

**Review by tqchen — COMMENTED** [review]
some minor nits

**Comment by MasterJH5574** [comment]
@tqchen Thanks! I have addressed these comments.

**Pull request merged**